### DISCUSSION TITLE
How to name branches, PR, and commit messages

### DISCUSSION THREAD

**Discussion by Sysix** [oxc-project/oxc#4670] [General]
Hello,

After working on my first `oxc_linter` rule, I had some trouble figuring out how I should name my branches, PRs, and commit messages.
I would love to see an update on [CONTRIBUTING.md](https://github.com/oxc-project/oxc/blob/main/CONTRIBUTING.md) or its link references.

**Branches**:
Does it really matter? I did not care, but I found out you are using a month-day prefix

**PR**:
When I'm adding a new rule, does the PR name call itself:

`feat(linter/<plugin-name>): implement <rule-name>` or   
`feat(linter) implement <plugin-name>/<rule-name>`

This looks like it will affect the release documents, so we should keep it consistent. See the last release, which has different implementations:
https://github.com/oxc-project/oxc/releases/tag/oxlint_v0.7.0

**Commit Messages**:
It looks like the PR commits are getting squashed. Do you have a preferred way to improve the commit messages to help with the Review?

**Comment by Boshen** [comment]
Branch name does not matter. month-day prefix is created from graphite automatically.

Commit message should be `feat(linter): implement <plugin-name>/<rule-name>`.

> Do you have a preferred way to improve the commit messages to help with the Review?

Nope, anything is fine.
